packages/runner/lib/sdk/locks.ts

```diff
@@ -2,62 +2,420 @@
 import type { Result } from '@nangohq/utils';
 import { Err, Ok } from '@nangohq/utils';
 
-interface Lock {
-    key: string;
-    owner: string;
-    expiresAt: Date;
-}
+const MAX_LOCKS = 256;
+const STATE_FREE = 0;
+const STATE_LOCKED = 1;
+const MAX_STRING_LEN = 255; // Max length for key and owner strings
```

Comment:
owner and key strings are fixed size which is a bit of wasted space, especially if most of the lock keys used are short but once again I didn't want to deal with arbitrary length